replica/compaction_group.hh

```diff
@@ -243,12 +244,13 @@ public:
         return _compaction_groups;
     }
 
-    const storage_group_vector& storage_groups() const noexcept {
+    const storage_group_map& storage_groups() const noexcept {
         return _storage_groups;
     }
-    storage_group_vector& storage_groups() noexcept {
+    storage_group_map& storage_groups() noexcept {
         return _storage_groups;
     }
+    storage_group* storage_group_for_id(const schema_ptr&, size_t i) const;
```

Comment:
Please document that it always returns a non-null pointer.
This FIXME comment for the table::storage_group_for_id entry point applies here too.

Reply:
Also, can be virtual, with a trivial implementation for the single group manager, but I don't know if it's worth it.